### ISSUE TITLE
zsh auto-complete is partialy borken

### ISSUE THREAD

**Issue by chrisWhyTea-zz** [hetznercloud/cli#81]
Just builded the client on a ubuntu-16.04 machine (while being connectec via ssh) with golang-1.9 the auto-completion sometimes fails (for example auto completing the ssh-key inside the server create) will lead to the following error:
`_handle_flag:27: bad math expression: illegal character: ^[`

**Comment by thcyron** [comment]
@thetechnick Can you have a look?

**Comment by chrisWhyTea** [comment]
**How to reproduce:**
- use windows 10 with git bash
- ssh to a linux machine with installed zsh and golang-1.9
- install hcloud lib via `go get` and configure the autocompletion for zsh
- create and use a hcloud context
- try to create a new server `hcloud server create --name xyz.xyz --ssh-key` now try to autocomplete the sshkey

**Result:**
`_handle_flag:27: bad math expression: illegal character: ^[`

**Comment by hbokh** [comment]
FWIW - this seems to be of no issue on macOS 10.13.3 with `zsh 5.4.2 (x86_64-apple-darwin17.3.0)` / "Oh-My-Zsh".

**Comment by joto** [comment]
I can reproduce this on Linux with zsh 5.3.1. The culprit seem to be these lines:

```
    # keep flag value with flagname as flaghash
    # flaghash variable is an associative array which is only supported in bash > 3.
    if [[ -z "${BASH_VERSION}" || "${BASH_VERSINFO[0]}" -gt 3 ]]; then
        if [ -n "${flagvalue}" ] ; then
            flaghash[${flagname}]=${flagvalue}
        elif [ -n "${words[ $((c+1)) ]}" ] ; then
            flaghash[${flagname}]=${words[ $((c+1)) ]}
        else
            flaghash[${flagname}]="true" # pad "true" for bool flag
        fi
    fi
```

It doesn't seem like the `flaghash` is used anywhere. It is set here but never read. I just removed those lines and the error went away.

**Comment by joto** [comment]
Ups. What did just happen. Why does it say "joto unassigned thetechnick a minute ago"? I didn't do anything like that and don't think I can even do that...

**Comment by joto** [comment]
Does that mean that this comment will "unstale" it?

**Comment by eike-fokken** [comment]
If it helps, I have a similar (the same?) issue directly on Linux with zsh 5.8 (x86_64-debian-linux-gnu)
I'm using the workaround for antigen described in #239 .

Error message when pressing tab after:
`hcloud server create --image debian-10 --ssh-key `
is

`hcloud server create --image debian-10 --ssh-key __hcloud_handle_flag:31: bad math expression: illegal character: ^[`

**Comment by LKaemmerling** [comment]
Closed with #239
